### ISSUE TITLE
A question on the dropout noise

### ISSUE THREAD

**Issue by 787629504** [karoly-hars/GAN_image_colorizing#2]
I'm sorry to disturb you. I found that in your code, there is no "model.train()" or " model.test". Therefore, dropout and BatchNormal are also used during the test. Does this have an impact on the results？

**Comment by karoly-hars** [comment]
Yeah, all the models in the repo are always in train mode. This only matters because of the dropout and the batch normalization layers. Sometimes, these techniques are used differently in GANs than in regular convolutional networks:
- If you are training a DCGAN, it is preferred to do the normalization using the mean and variance of the current batch instead of the running statistics. It will make the training process more stable.
- The advantage of using dropout is that it adds a little more randomness to the generator which can make the discriminator's job more difficult and this can help avoid mode collapse.

Some discussions about regularization techniques for GANs:
- https://discuss.pytorch.org/t/why-dont-we-put-models-in-train-or-eval-modes-in-dcgan-example/7422/2
- https://towardsdatascience.com/gan-ways-to-improve-gan-performance-acf37f9f59b

**Comment by 787629504** [comment]
Thank you very much for your reply. I have a small problem about discriminator, if the fake accuracy is large(85%) and real accuracy is not large(65%), what can i do for this problem. Thank you.

**Comment by karoly-hars** [comment]
I don't think that this is necessarily a problem. It just means that the discriminator is better at correctly classifying the real samples than the fake ones.

**Comment by 787629504** [comment]
> I don't think that this is necessarily a problem. It just means that the discriminator is better at correctly classifying the real samples than the fake ones.

Thank you very much for your guidance!
